AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/ReceiverHandler.java

```diff
@@ -138,22 +140,35 @@ private JFieldVar createReceiverField(HasReceiverRegistration holder, String rec
 	private void registerAndUnregisterReceiver(HasReceiverRegistration holder, Receiver.RegisterAt registerAt, JFieldVar intentFilterField, JFieldVar receiverField, boolean local) {
 		JBlock registerBlock = null;
 		JBlock unregisterBlock = null;
-		switch (registerAt) {
-		case OnCreateOnDestroy:
-			registerBlock = holder.getOnCreateAfterSuperBlock();
-			unregisterBlock = holder.getOnDestroyBeforeSuperBlock();
-			break;
-		case OnStartOnStop:
-			registerBlock = holder.getOnStartAfterSuperBlock();
-			unregisterBlock = holder.getOnStopBeforeSuperBlock();
-			break;
-		case OnResumeOnPause:
-			registerBlock = holder.getOnResumeAfterSuperBlock();
-			unregisterBlock = holder.getOnPauseBeforeSuperBlock();
-			break;
-		case OnAttachOnDetach:
-			registerBlock = holder.getOnAttachAfterSuperBlock();
-			unregisterBlock = holder.getOnDetachBeforeSuperBlock();
+
```

Comment:
This code is now rather ugly, but this is the price for removing the invalid abstraction `HasLifecycleMethods`. Do you have a better idea? @dodgex

Reply:
Currently i cant think of a better way to solve this. :/